backend/routes.js

```diff
@@ -66,6 +66,21 @@ route.GET('/events/{contractID}/{since}', {}, async function (request, h) {
   }
 })
 
+route.GET('/eventsBefore/{contractID}', {}, async function (request, h) {
```

Comment:
- This doesn't need `contractID` in the parameters.
- We must create a uniform and consistent interface for our backend API. If you think switching to query parameters is better (and I'm inclined to agree), then you need to argue why it's better, and also update the other APIs to be consistent. For example, you could say that query parameters are more natural for this use-case and are better supported across different libraries and frameworks. I would believe that, as it's true (@snowteamer do you agree with that reasoning? I know you're working on Deno, so your input would be valuable here).
- And so, if we switch to query parameters, we also need to change the `'/events/{contractID}/{since}'` to be consistent. So for example, we'd change that API to be `/eventsSince?contractID={cID}&since={since}` and update all of the code accordingly.

Reply:
I'm not sure which is better between query parameters and route parameters. Queray parameters are likely easier to implement in vanilla, but even a simple router will usually support route parameters too.

[This post](https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api) might give some insights.